packages/playwright/src/runner/dispatcher.ts

```diff
@@ -361,6 +362,8 @@ class JobDispatcher {
       body: params.body !== undefined ? Buffer.from(params.body, 'base64') : undefined
     };
     data.result.attachments.push(attachment);
+    if (params.stepId)
+      data.steps.get(params.stepId)!.attachments.push(attachment);
```

Comment:
Let's not crash when step is not found, but print an internal error instead, see line 340.

Reply:
done in https://github.com/microsoft/playwright/pull/34037/commits/ba69c19baa42bd0a34f24bd4dd9da8bdc5fd2a8e